Feat: In app notifications

davecreaser opened the pull request:
In app notifications full feature.

Auth proxy: https://github.com/JoinColony/colonyCDappAuthProxy/pull/31
Block Ingestor: https://github.com/JoinColony/block-ingestor/pull/287

jakubcolony reviewed (approved):
Very nice, well done @davecreaser and everyone else who contributed! 🎉 

Had a quick look through the PRs and all looks good. When I try to run the branch locally, I'm still having the issue I noted in #3400:
> I had a bit of a trouble getting the notifications dev key to generate. After I switched branches and ran npm run prepare, there was no entry for MAGICBELL_DEV_KEY in .env so the script could not replace it properly. I had to add that var with empty value to .env.local and run prepare again for it to work.

What are your thoughts on the above? Should the dev key entry be included in the `.env.local` as well? Or maybe the script can append it to the file if it's not there already?

davecreaser commented:
> Very nice, well done @davecreaser and everyone else who contributed! 🎉
> 
> Had a quick look through the PRs and all looks good. When I try to run the branch locally, I'm still having the issue I noted in #3400:
> 
> > I had a bit of a trouble getting the notifications dev key to generate. After I switched branches and ran npm run prepare, there was no entry for MAGICBELL_DEV_KEY in .env so the script could not replace it properly. I had to add that var with empty value to .env.local and run prepare again for it to work.
> 
> What are your thoughts on the above? Should the dev key entry be included in the `.env.local` as well? Or maybe the script can append it to the file if it's not there already?

Yeah thanks for raising this again Jakub - I've added all the notifications vars to .env.local 😄
